### PR TITLE
[CSL-948] Automatically fix permissions on keyfile

### PR DESCRIPTION
**Pull request by Pastafarianist**

### CODE REVIEW

**Comment by gromakovsky** [comment]
I am 99% sure we need it in 0.4. So you should:
1. Squash commits.
2. Cherry-pick commit to be on top of 0.4 HEAD.
3. Make a PR from that commit (cherry-picked on top of 0.4) to 0.4 branch.

After that we will merge 0.4 to master.

File: src/keygen/Testnet.hs [hunk 1]
```diff
@@ -20,16 +20,19 @@ import           Pos.Util.UserSecret  (initializeUserSecret, takeUserSecret, usP
 
 import           KeygenOptions        (TestStakeOptions (..))
 
-generateKeyfile :: FilePath -> IO (PublicKey, VssCertificate)
+import           System.Wlog          (WithLogger)
+
+generateKeyfile :: (MonadIO m, MonadFail m, WithLogger m) => FilePath -> m (PublicKey, VssCertificate)
```
Comment (gromakovsky) [inline src/keygen/Testnet.hs]: This line is too long.

File: src/keygen/Testnet.hs [hunk 2]
```diff
@@ -20,16 +20,19 @@ import           Pos.Util.UserSecret  (initializeUserSecret, takeUserSecret, usP
...
 generateKeyfile fp = do
     initializeUserSecret fp
     sk <- snd <$> keyGen
     vss <- vssKeyGen
     us <- takeUserSecret fp
+    -- liftIO $ do
```
Comment (gromakovsky) [inline src/keygen/Testnet.hs]: 🤔

File: src/node/Main.hs [hunk 3]
```diff
@@ -186,7 +186,7 @@ processUserSecret args@Args {..} userSecret = case backupPhrase of
         writeUserSecret us
         return (sk, us)
 
-getNodeParams :: (MonadIO m, MonadFail m, MonadThrow m) => Args -> Timestamp -> m NodeParams
+getNodeParams :: (MonadIO m, MonadFail m, MonadThrow m, WithLogger m) => Args -> Timestamp -> m NodeParams
```
Comment (gromakovsky) [inline src/node/Main.hs]: This line is too long.

**Comment by gromakovsky** [comment]
DId you check it works, btw?

**Comment by Pastafarianist** [comment]
@gromakovsky Yes, I tested it.

**Comment by Pastafarianist** [comment]
@gromakovsky also I've cherry-picked this PR to `cardano-0.4-rc` as you specified.

**Comment by gromakovsky** [comment]
@Pastafarianist ok, now you need to merge `cardano-sl-0.4-rc` into your branch because there are conflicts. Or you can rebase if you want.

**Pull request merged**